src/main/scala/com/microsoft/hyperspace/index/IndexLogManager.scala

```diff
@@ -71,7 +72,13 @@ class IndexLogManagerImpl(indexPath: Path) extends IndexLogManager with Logging
       return None
     }
     val contents = FileUtils.readContents(fs, path)
-    Some(LogEntry.fromJson(contents))
+    Try(LogEntry.fromJson(contents)) match {
+      case Success(s) => Some(s)
+      case Failure(NonFatal(e)) =>
+        logError(s"Error reading LogEntry json from $path, $e")
+        None
+      case Failure(e) => throw e
```

Comment:
From this it looks like if `contents` is corrupted, and json conversion fails, it will not throw the error as it is Non-Fatal.
Could you change it to throw in case of all failures? We would not want file to be present but unreadable.

cc @imback82

Reply:
right, this change should be just reverted as mentioned here: https://github.com/microsoft/hyperspace/pull/92#discussion_r456736210

Reply:
ok reverted, but it does not look like valid code :
We are returning ```Some``` of something that can throw, but we can essentially never get a ```None``` here